src/controllers/confirm.company.controller.ts

```diff
@@ -47,7 +49,9 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const token: string = retrieveAccessTokenFromSession(session);
     const company: ObjectionCompanyProfile = retrieveCompanyProfileFromObjectionSession(session);
-    const objectionId = await createNewObjection(company.companyNumber, token);
+    const objectionCreate: ObjectionCreate = retrieveObjectionCreateFromObjectionSession(session);
+    deleteObjectionCreateFromObjectionSession(session);
+    const objectionId = await createNewObjection(company.companyNumber, token, objectionCreate);
     addToObjectionSession(session, SESSION_OBJECTION_ID, objectionId);
     return res.redirect(OBJECTIONS_ENTER_INFORMATION);
   } catch (e) {
```

Comment:
I think we should also delete the objection create from the session if we catch an exception. This could occur due to access token missing or company profile missing.

Reply:
not sure if there is a finally block in js that we could put the delete in.

Reply:
Decided not to use a finally block as tyhere is a return in the catch, instead just made sure is the first function carried out.